pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -345,39 +359,87 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
+	if err != nil {
+		msg := fmt.Sprintf("Failed to get the list of clients VMs for IP %q. Error: %+v", tkgVMIP, err)
 		log.Error(msg)
 		return errors.New(msg)
 	}
+	if len(clientVms) == 0 {
+		return nil
+	}
+	if len(clientVms) == 1 && clientVms[0] == vm.Name {
+		// Prepare the CnsVolumeACLConfigureSpec
+		cnsVolumeID := cnstypes.CnsVolumeId{
+			Id: volumeID,
+		}
+		vSanFileShareNetPermissions := make([]vsanfstypes.VsanFileShareNetPermission, 0)
+		vsanFileShareAccessType := vsanfstypes.VsanFileShareAccessTypeREAD_WRITE
+		vSanFileShareNetPermissions = append(vSanFileShareNetPermissions, vsanfstypes.VsanFileShareNetPermission{
+			Ips:         tkgVMIP,
+			Permissions: vsanFileShareAccessType,
+		})
 
-	tkgVMIP, err := cnsoperatorutil.GetTKGVMIP(ctx, r.vmOperatorClient, r.dynamicClient, vm.Namespace, vm.Name, nsxConfiguration)
+		cnsNFSAccessControlSpecList := make([]cnstypes.CnsNFSAccessControlSpec, 0)
+		cnsNFSAccessControlSpecList = append(cnsNFSAccessControlSpecList, cnstypes.CnsNFSAccessControlSpec{
+			Permission: vSanFileShareNetPermissions,
+			Delete:     true,
+		})
+
+		cnsVolumeACLConfigSpec := cnstypes.CnsVolumeACLConfigureSpec{
+			VolumeId:              cnsVolumeID,
+			AccessControlSpecList: cnsNFSAccessControlSpecList,
+		}
+		err = r.volumeManager.ConfigureVolumeACLs(ctx, cnsVolumeACLConfigSpec)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to configure ACLs for volume: %q. Error: %+v", volumeID, err)
+			log.Error(msg)
+			return errors.New(msg)
+		}
+		log.Debugf("Successfully configured ACLs for volume %q", volumeID)
+	}
+	err = cnsFileVolumeClientInstance.RemoveClientVMFromIPList(ctx, instance.Namespace+"/"+instance.Spec.PvcName, instance.Spec.VMName, tkgVMIP)
 	if err != nil {
-		msg := fmt.Sprintf("Failed to get external facing IP address for VM %q/%q. Err: %+v", vm.Namespace, vm.Name, err)
+		msg := fmt.Sprintf("Failed to remove VM %q with IP %q to IPList. Error: %+v", vm.Name, tkgVMIP, err)
 		log.Error(msg)
 		return errors.New(msg)
 	}
+	log.Debugf("Successfully removed VM IP %q to IPList for CnsFileAccessConfig request with name: %q on namespace: %q", tkgVMIP, instance.Name, instance.Namespace)
+	return nil
+}
 
-	log.Debugf("Found tkg VMIP %q for VM %q in namespace %q", tkgVMIP, vm.Name, vm.Namespace)
+// configureFileVolumeACL helps to add net permissions for file volumes by invoking CNS API
+// This method will add the vm entry in ExternalIPtoClientVms for a given cnsfilevolumeclient instance
+//  Returns error if any operation fails
+func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACL(ctx context.Context, volumeID string, vm *vmoperatortypes.VirtualMachine, instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig) error {
+	log := logger.GetLogger(ctx)
+	tkgVMIP, err := r.getExternalIP(ctx, vm)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to get CNSFileVolumeClient instance. Error: %+v", err)
```

Comment:
Change the error message -> something like 'failed to get external IP for the vm. Error: +%v'

Reply:
Done